app/controllers/users/sessions_controller.rb

```diff
@@ -79,6 +80,40 @@ def process_locked_out_session
       redirect_to root_url
     end
 
+    def valid_captcha_result?
+      if cookies[:device] && (device = Device.find_by(cookie_uuid: cookies[:device]))
+        return true if device.user.email_addresses.lazy.map(&:email).include?(auth_params[:email])
```

Comment:
oooo smart use of `.lazy`! Another idea:
```suggestion
        return true if device.user.email_addresses.any? { |e| e.email == auth_params[:email] }
```

Reply:
Yeah I thought it was a nice way to avoid having to use a block, but funny that the block form ends up being shorter anyways 😅